src/fsharp/ConstraintSolver.fs

```diff
@@ -1552,43 +1542,42 @@ and AddConstraint (csenv:ConstraintSolverEnv) ndeep m2 trace tp newConstraint  =
 
-        | TyparConstraint.IsDelegate(aty1,bty1,_),TyparConstraint.IsDelegate(aty2,bty2,_) -> 
+        | TyparConstraint.IsDelegate(aty1,bty1,_), TyparConstraint.IsDelegate(aty2,bty2,_) -> 
             typeEquiv g aty1 aty2 && typeEquiv g bty1 bty2 
 
-        | TyparConstraint.SupportsComparison _,TyparConstraint.SupportsComparison _  
-        | TyparConstraint.SupportsEquality _,TyparConstraint.SupportsEquality _  
+        | TyparConstraint.SupportsComparison _, TyparConstraint.SupportsComparison _
+        | TyparConstraint.SupportsEquality _, TyparConstraint.SupportsEquality _
         // comparison implies equality
-        | TyparConstraint.SupportsComparison _,TyparConstraint.SupportsEquality _  
-        | TyparConstraint.SupportsNull _,TyparConstraint.SupportsNull _  
-        | TyparConstraint.IsNonNullableStruct _,TyparConstraint.IsNonNullableStruct _     
+        | TyparConstraint.SupportsComparison _, TyparConstraint.SupportsEquality _
+        | TyparConstraint.SupportsNull _, TyparConstraint.SupportsNull _
+        | TyparConstraint.IsNonNullableStruct _, TyparConstraint.IsNonNullableStruct _
         | TyparConstraint.IsUnmanaged _, TyparConstraint.IsUnmanaged _
-        | TyparConstraint.IsReferenceType _,TyparConstraint.IsReferenceType _ 
-        | TyparConstraint.RequiresDefaultConstructor _,TyparConstraint.RequiresDefaultConstructor _ -> true
-        | TyparConstraint.SimpleChoice (tys1,_),TyparConstraint.SimpleChoice (tys2,_) -> ListSet.isSubsetOf (typeEquiv g) tys1 tys2
+        | TyparConstraint.IsReferenceType _, TyparConstraint.IsReferenceType _
+        | TyparConstraint.RequiresDefaultConstructor _, TyparConstraint.RequiresDefaultConstructor _ -> true
+        | TyparConstraint.SimpleChoice (tys1,_), TyparConstraint.SimpleChoice (tys2,_) -> ListSet.isSubsetOf (typeEquiv g) tys1 tys2
         | TyparConstraint.DefaultsTo (priority1,dty1,_), TyparConstraint.DefaultsTo (priority2,dty2,_) -> 
              (priority1 = priority2) && typeEquiv g dty1 dty2
         | _ -> false
-
         
     
     // First ensure constraint conforms with existing constraints 
     // NOTE: QUADRATIC 
     let existingConstraints = tp.Constraints
 
     let allCxs = newConstraint :: List.rev existingConstraints
-    begin 
-        let rec enforceMutualConsistency i cxs = 
-            match cxs with 
-            | [] ->  CompleteD
-            | cx :: rest -> IterateIdxD (fun j cx2 -> if i = j then CompleteD else consistent cx cx2) allCxs ++ (fun () -> enforceMutualConsistency (i+1) rest)
-
-        enforceMutualConsistency 0 allCxs 
-    end ++ (fun ()  ->
+     
+    let rec enforceMutualConsistency i cxs = 
+        match cxs with 
+        | [] ->  CompleteD
+        | cx :: rest -> IterateIdxD (fun j cx2 -> if i = j then CompleteD else consistent cx cx2) allCxs ++ (fun () -> enforceMutualConsistency (i+1) rest)
+
+    enforceMutualConsistency 0 allCxs 
```

Comment:
Again explicit parentheses to clarify ``(a @ b) ++ c`` rather than ``a @ b ++ c`` independent of whitespace rules please